storage/api/Storage.Samples/DownloadPublicFile.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START storage_download_public_file]
+
+using Google.Cloud.Storage.V1;
+using System;
+using System.IO;
+
+public class DownloadPublicFile
+{
+	public string Download(string bucketName = "your-bucket-name", string objectName = "your-object-name", string localPath = null)
+	{
+		var storage = StorageClient.CreateUnauthenticated();
+
+		localPath = localPath ?? Path.GetFileName(objectName);
```

Comment:
As We are passing full path we don't need this.
Note: this Applies to all other samples as well,  refer the `DownloadFileSample` sample.